src/wp-includes/media.php

```diff
@@ -1911,6 +1918,36 @@ function wp_img_tag_add_loading_attr( $image, $context ) {
 	return $image;
 }
 
+/**
+ * Adds `decoding=async` attribute to an `img` HTML tag.
+ *
+ * @since 6.0.0
+ *
+ * @param string $image   The HTML `img` tag where the attribute should be added.
+ * @param string $context Additional context to pass to the filters.
+ * @return string Converted `img` tag with `decoding` attribute added.
+ */
+function wp_img_tag_add_decoding_async_attr( $image, $context ) {
+	/**
+	 * Filters the `decoding` attribute value to add to an image. Default `async`.
+	 *
+	 * Returning `false` or an empty string will not add the attribute.
+	 *
+	 * @since 6.0.0
+	 *
+	 * @param string|bool $value   The `decoding` attribute value. Returning a falsey value will result in
+	 *                             the attribute being omitted for the image.
+	 * @param string      $image   The HTML `img` tag to be filtered.
+	 * @param string      $context Additional context about how the function was called or where the img tag is.
+	 */
+	$value = apply_filters( 'wp_img_tag_add_decoding_attr', 'async', $image, $context );
+	if ( ! $value || ! in_array( $value, array( 'async', 'sync' ), true ) ) {
+		return $image;
+	}
+
+	return str_replace( '<img ', '<img decoding="' . esc_attr( $value ) . '" ', $image );
```

Comment:
I just realized that the value of `auto` is also allowed: https://developer.mozilla.org/en-US/docs/Web/API/HTMLImageElement/decoding

This accounts for that and reduces the `return` count to 1.

```suggestion
	if ( $value && in_array( $value, array( 'async', 'sync', 'auto' ), true ) ) {
		$image = str_replace( '<img ', '<img decoding="' . esc_attr( $value ) . '" ', $image );
	}

	return $image;
```

Reply:
Ah, good catch!